Sorry for long delay, I approved it long time ago internally, but then lost track of it. It failed to submit because of compilation errors:

```tensorflow/core/grappler/utils_test.cc:426:20: error: extra tokens at end of #ifdef directive [-Werror,-Wextra-tokens]
#ifdef GOOGLE_CUDA || TENSORFLOW_USE_ROCM```\

This should be `#if defined(...)`

--- a/tensorflow/core/grappler/utils_test.cc
+++ b/tensorflow/core/grappler/utils_test.cc
@@ -423,7 +423,7 @@ TEST(IsKernelRegisteredForNode, All) {
   v.set_type(DataType::DT_FLOAT);
   (*node.mutable_attr())["T"] = v;
   TF_EXPECT_OK(IsKernelRegisteredForNode(node));
-#ifdef GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
   node.set_device("/gpu:0");
   TF_EXPECT_OK(IsKernelRegisteredForNode(node));
 #endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM

